scene/gui/text_edit.h

```diff
@@ -883,20 +888,26 @@ class TextEdit : public Control {
 	void select_all();
 	void select_word_under_caret(int p_caret = -1);
 	void add_selection_for_next_occurrence();
-	void select(int p_from_line, int p_from_column, int p_to_line, int p_to_column, int p_caret = 0);
+	void select(int p_origin_line, int p_origin_column, int p_caret_line, int p_caret_column, int p_caret = 0);
 
 	bool has_selection(int p_caret = -1) const;
 
 	String get_selected_text(int p_caret = -1);
+	int get_selection_at(int p_line, int p_column, bool p_include_edges = true) const;
+	Vector<Point2i> get_line_ranges_from_carets(bool p_only_selections = false, bool p_merge_adjacent = true) const;
 
-	int get_selection_line(int p_caret = 0) const;
-	int get_selection_column(int p_caret = 0) const;
+	void set_selection_origin_line(int p_line, int p_caret = 0, bool p_can_be_hidden = true, int p_wrap_index = -1);
```

Comment:
This does make it more awkward, since the `p_caret` param is always used, and the other two are only sometimes used. With `p_caret` at the end they wouldn't even need to be optional.
But I guess if consistency is more important, I'll change it.